File: navigation/src/main/java/com/example/navigation/stepsEngine/field/MustCollectSignatureEMVField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class MustCollectSignatureEMVField extends Field {

    @Override
    public Boolean get(FlowState flowState) {
        return flowState.getCard().isMustCollectSignatureEMV();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/MustCollectSignatureEMVField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState

class MustCollectSignatureEMVField : Field() {
    override operator fun get(flowState: FlowState): Boolean? {
        return flowState.card.isMustCollectSignatureEMV
    }

    override fun getId(): String {
        return FieldName.MUST_COLLECT_SIGNATURE_EMV
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}